app/api/controllers/validators/resolvers/ResolveDateRange.scala

```diff
@@ -27,6 +27,9 @@ case class DateRange(startDate: LocalDate, endDate: LocalDate)
 
 object ResolveDateRange extends Resolver[(String, String), DateRange] {
 
+  private val minYear : Int = 1900
```

Comment:
These changes shouldn't really be going into ResolveDateRange, as it's common code. The minYear and maxYear stuff is kind of specific to this API & endpoint.
You could just add a couple of functions into the ValidatorFactory class that return Either[Seq[Invalid], Valid].